File: Android/doraemonkit/src/main/java/com/didichuxing/doraemonkit/util/ColorUtil.java
package com.didichuxing.doraemonkit.util;

import android.graphics.Color;
import androidx.annotation.ColorInt;

/**
 * Created by wanglikun on 2018/9/15.
 */

public class ColorUtil {

    private ColorUtil() {
    }

    public static String parseColorInt(@ColorInt int color) {
        return String.format("#%06X", 0xFFFFFF & color);
    }

    public static boolean isColdColor(@ColorInt int color) {
        float[] hsv = new float[3];
        Color.colorToHSV(color, hsv);
        return hsv[2] <= 0.8f;
    }
}

File: Android/kotlin/doraemonkit/src/main/java/com/didichuxing/doraemonkit/util/ColorUtil.kt
package com.didichuxing.doraemonkit.util

import android.graphics.Color
import androidx.annotation.ColorInt

/**
 * Created by wanglikun on 2018/9/15.
 */
object ColorUtil {
    fun parseColorInt(@ColorInt color: Int): String {
        return String.format("#%06X", 0xFFFFFF and color)
    }

    fun isColdColor(@ColorInt color: Int): Boolean {
        val hsv = FloatArray(3)
        Color.colorToHSV(color, hsv)
        return hsv[2] <= 0.8f
    }
}